tgg/relatedness/n_non_ref_pairs.py

```diff
@@ -0,0 +1,273 @@
+        f"{temp_path}/n_non_ref_{non_ref_samples}_sample_high_quality_sites.ht",
+        overwrite=True,
+    )
+    return ht
+
+
+def get_and_count_sample_pairs(
+    mt: hl.MatrixTable, temp_path=TEMP_PATH, non_ref_samples=3,
+) -> hl.Table:
+    """
+    Return the number of shared n non_ref sites per pair.
+
+    :param mt: Matrix Table to compute pairs on.
+    :param temp_path: Path to write pair HT to. Default to TEMP_PATH.
+    :param non_ref_samples: Number of non_ref samples per site. Defaults to 3.
+    :return ht: MatrixTable
+    """
+    logger.info("Collecting samples and counting sample pairs...")
+    mt = mt.annotate_rows(
+        samples=hl.agg.filter(mt.GT.is_non_ref(), hl.agg.collect(mt.s))
+    )
+    mt = mt.annotate_rows(
+        sample_pairs=hl.range(0, mt.samples.length()).flatmap(
+            lambda i: hl.range(i + 1, mt.samples.length()).map(
+                lambda j: hl.tuple([mt.samples[i], mt.samples[j]])
+            )
+        )
+    )
+    ht = mt.select_rows(mt.sample_pairs).rows()
+    ht = ht.explode(ht.sample_pairs)
+
+    logger.info("Aggregating shared sites per pair...")
+    ht = ht.group_by(ht.sample_pairs).aggregate(n_non_ref_sites_shared=hl.agg.count())
+    ht = ht.checkpoint(
+        f"{temp_path}/pairwise_shared_{non_ref_samples}_sites.ht", overwrite=True
+    )
+    return ht
+
+
+def get_samples_n_non_ref(
+    vds_path: str = VDS_PATH,
+    temp_path: str = TEMP_PATH,
+    control_samples: Set[str] = {NA12878, SYNDIP},
+    non_ref_samples: int = 3,
+    het_only: bool = False,
+):
+    """
+    Get number of shared non ref sites per sample pair in the 455k VDS.
+
+    Filter VDS variant data to sites present in specified input Table, collect sample IDs, annotate IDs onto rows,
+    explode on sample pairs, count pairs and write HT to temporary path.
+
+    :param vds_path: Path to UKB 455k VDS. Default is VDS_PATH.
+    :param temp_path: Path to bucket to store Table and other temporary data. Default is TEMP_PATH.
+    :param control_samples: Set of control sample IDs to remove. Default is {NA12878, SYNDIP}.
+    :param non_ref_samples: Number of non_ref samples per site to filter to. Defaults to 3.
+    :param het_only: Filter the non ref sample sites to those with only het sample calls. Defaults to False.
+    :return: Table keyed by sample IDs and their number of singletons.
+    """
+    logger.info(
+        "Retrieving pairwise shared %i non ref sample sites...", non_ref_samples
+    )
+    mt = hl.vds.read_vds(vds_path).variant_data
+    mt = mt.filter_cols(~hl.literal(control_samples).contains(mt.s))
+    ht = get_n_non_ref_sites(
+        vds_path=vds_path,
+        temp_path=temp_path,
+        non_ref_samples=non_ref_samples,
+        het_only=het_only,
+    )
+    mt = mt.annotate_rows(**ht[mt.row_key])
+    mt = mt.filter_rows(hl.is_defined(mt.ac))
+    ht = get_and_count_sample_pairs(
+        mt, temp_path=temp_path, non_ref_samples=non_ref_samples
+    )
+    return ht
```

Comment:
```suggestion
    get_and_count_sample_pairs(
        mt, temp_path=temp_path, non_ref_samples=non_ref_samples
    )
```
you don't need to return this HT since the code isn't doing anything with it downstream, right? this is also true of `get_and_count_sample_pairs` (could return None and just write table to temp path)

Reply:
good point!